addons/saml-sp-metadata.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" entityID="[% sp_entity_id %]">
+  <md:SPSSODescriptor protocolSupportEnumeration="urn:oasis:names:tc:SAML:1.1:protocol urn:oasis:names:tc:SAML:2.0:protocol">
+    <md:KeyDescriptor use="signing">
+      <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
+        <ds:X509Data>
+          <ds:X509Certificate>[% sp_cert %]</ds:X509Certificate>
+        </ds:X509Data>
+      </ds:KeyInfo>
+    </md:KeyDescriptor>
+    <md:KeyDescriptor use="encryption">
+      <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
+        <ds:X509Data>
+          <ds:X509Certificate>[% sp_cert %]</ds:X509Certificate>
+        </ds:X509Data>
+      </ds:KeyInfo>
+    </md:KeyDescriptor>
+    <!--- Not implemented and probably will never be... Leaving it there should it prove to be mandatory for a SAML IDP
+    <md:SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="[% protocol %]://[% hostname %]/saml/assertion"/>
+    -->
+    <md:AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="[% protocol %]://[% hostname %]/saml/assertion" index="0"/>
+    <md:AssertionConsumerService Binding="urn:oasis:names:tc:SAML:1.0:profiles:browser-post" Location="[% protocol %]://[% hostname %]/saml/assertion" index="1"/>
+    <md:AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact" Location="[% protocol %]://[% hostname %]/saml/assertion" index="2"/>
+    <md:AssertionConsumerService Binding="urn:oasis:names:tc:SAML:1.0:profiles:artifact-01" Location="[% protocol %]://[% hostname %]/saml/assertion" index="3"/>
+  </md:SPSSODescriptor>
+</md:EntityDescriptor>
+
```

Comment:
Use the xml filter for all output
[% sp_entity_id | xml %]
Or use Template::AutoFilter 
my $tt = Template::AutoFilter->new( AUTO_FILTER => 'xml' );

http://search.cpan.org/~mithaldu/Template-AutoFilter/lib/Template/AutoFilter.pm